rs/index/src/multi_spann/builder.rs

```diff
@@ -1,16 +1,27 @@
-use std::sync::RwLock;
+use std::sync::atomic::{AtomicU64, Ordering};
 
 use anyhow::Result;
 use config::collection::CollectionConfig;
 use dashmap::mapref::entry::Entry;
 use dashmap::DashMap;
+use lock_api::RwLockUpgradableReadGuard;
 use log::debug;
+use parking_lot::RwLock;
+use utils::bloom_filter::blocked_bloom_filter::BlockedBloomFilter;
 
 use crate::spann::builder::{SpannBuilder, SpannBuilderConfig};
 
+#[derive(Hash)]
+pub struct BloomFilterKey {
+    user_id: u128,
+    doc_id: u128,
+}
+
 pub struct MultiSpannBuilder {
     config: CollectionConfig,
     inner_builders: DashMap<u128, RwLock<SpannBuilder>>,
+    bloom_filter: RwLock<Option<BlockedBloomFilter>>,
```

Comment:
why do you  need `RwLock` here? I don't think we build or write in multithreaded right?